sdk/edgezones/arm-edgezones/review/arm-edgezones.api.md

```diff
@@ -0,0 +1,212 @@
+    readonly code?: string;
+    readonly details?: ErrorDetail[];
+    readonly message?: string;
+    readonly target?: string;
+}
+
+// @public
+export interface ErrorResponse {
+    error?: ErrorDetail;
+}
+
+// @public
+export interface ExtendedZone extends ProxyResource {
+    properties?: ExtendedZoneProperties;
+}
+
+// @public
+export interface ExtendedZoneProperties {
+    readonly displayName: string;
+    readonly geography: string;
+    readonly geographyGroup: string;
+    readonly homeLocation: string;
+    readonly latitude: string;
+    readonly longitude: string;
+    readonly provisioningState?: ProvisioningState;
+    readonly regionalDisplayName: string;
+    readonly regionCategory: string;
+    readonly regionType: string;
+    readonly registrationState?: RegistrationState;
+}
+
+// @public
+export interface ExtendedZonesGetOptionalParams extends OperationOptions {
+}
+
+// @public
+export interface ExtendedZonesListBySubscriptionOptionalParams extends OperationOptions {
+}
+
+// @public
+export interface ExtendedZonesOperations {
+    get: (extendedZoneName: string, options?: ExtendedZonesGetOptionalParams) => Promise<ExtendedZone>;
+    listBySubscription: (options?: ExtendedZonesListBySubscriptionOptionalParams) => PagedAsyncIterableIterator<ExtendedZone>;
+    register: (extendedZoneName: string, options?: ExtendedZonesRegisterOptionalParams) => Promise<ExtendedZone>;
+    unregister: (extendedZoneName: string, options?: ExtendedZonesUnregisterOptionalParams) => Promise<ExtendedZone>;
+}
+
+// @public
+export interface ExtendedZonesRegisterOptionalParams extends OperationOptions {
+}
+
+// @public
+export interface ExtendedZonesUnregisterOptionalParams extends OperationOptions {
+}
+
+// @public
+export enum KnownActionType {
+    Internal = "Internal"
+}
+
+// @public
+export enum KnownCreatedByType {
+    Application = "Application",
+    Key = "Key",
+    ManagedIdentity = "ManagedIdentity",
+    User = "User"
+}
+
+// @public
+export enum KnownOrigin {
+    "user,system" = "user,system",
+    system = "system",
+    user = "user"
+}
+
+// @public
+export enum KnownRegistrationState {
+    NotRegistered = "NotRegistered",
+    PendingRegister = "PendingRegister",
+    PendingUnregister = "PendingUnregister",
+    Registered = "Registered"
+}
+
+// @public
+export enum KnownResourceProvisioningState {
+    Canceled = "Canceled",
+    Failed = "Failed",
+    Succeeded = "Succeeded"
+}
+
+// @public
+export interface Operation {
+    actionType?: ActionType;
+    display?: OperationDisplay;
+    readonly isDataAction?: boolean;
+    readonly name?: string;
+    readonly origin?: Origin;
+}
+
+// @public
+export interface OperationDisplay {
+    description?: string;
+    operation?: string;
+    provider?: string;
+    resource?: string;
+}
+
+// @public
+export interface OperationsListOptionalParams extends OperationOptions {
+}
+
+// @public
+export interface OperationsOperations {
```

Comment:
@qiaozha @joheredi is this name wired? should we rename this?

Reply:
I don't think this interface will be really used by our customers. And renaming is risky because of potential operation name and our own operation group name conflicts. I would say we should keep this way unless we get further customer feedback about this.